### PR TITLE
[Native][ACR-1644]update ml_training api descriptions

### PR DESCRIPTION
**Pull request by zhoonit**
Update ml_training api descriptions reflecting changes in acr-1644

Co-authored-by: hyeonseok lee <hs89.lee@samsung.com>
Signed-off-by: Jihoon Lee <jhoon.it.lee@samsung.com>

### Change Description ###

 - Reflect changes in ACR-1644.
 - Add default value column in layer property description

### API Changes ###

 - ACR: [ACR-1644](https://code.sec.samsung.net/jira/browse/ACR-1644)

### CODE REVIEW

File: docs/application/native/guides/machine-learning/machine-learning-training.md [hunk 1]
```diff
@@ -74,101 +74,230 @@ ml_train_model_construct(&model);
 // Compile model. This freezes model and afterwards the model cannot be modified.
```
Comment (annie-samsung) [inline docs/application/native/guides/machine-learning/machine-learning-training.md:74]: // Compile model - This freezes the model and afterward, the model cannot be modified.

File: docs/application/native/guides/machine-learning/machine-learning-training.md [hunk 2]
```diff
@@ -74,101 +74,230 @@ ml_train_model_construct(&model);
...
 ml_train_model_compile(model, "loss=cross", "batch_size=16", NULL);
 // run model
```
Comment (annie-samsung) [inline docs/application/native/guides/machine-learning/machine-learning-training.md:76]: // Run model

File: docs/application/native/guides/machine-learning/machine-learning-training.md [hunk 3]
```diff
@@ -74,101 +74,230 @@ ml_train_model_construct(&model);
...
-ml_train_model_run(model, "epochs=2", "save_path=model.bin", NULL);
+ml_train_model_run(model, "epochs=2", NULL);
+
+/// Save, loading in below format supported since Tizen 6.5.
```
Comment (annie-samsung) [inline docs/application/native/guides/machine-learning/machine-learning-training.md:79]:
```suggestion
// Save and load in the following format supported since Tizen 6.5:
```

File: docs/application/native/guides/machine-learning/machine-learning-training.md [hunk 4]
```diff
@@ -74,101 +74,230 @@ ml_train_model_construct(&model);
...
-Function | Key | Value | Description
--------- | --- | ----- | -----------
-`ml_train_model_compile()` | loss      | (categorical)| Loss function to be set
-&#xfeff;                 |           | cross        | cross-entropy loss
-&#xfeff;                 |           | mse          | mean squared error
-`ml_train_model_run()`     | epochs    | (integer)    | Determines epochs for the model
-&#xfeff;                 | save_path | (file_path)  | Model path to save and read parameters
+| Function                   | Key       | Value         | Description                            |
+| -------------------------- | --------- | ------------- | -------------------------------------- |
+| `ml_train_model_run()`     | epochs    | (integer)     | Determines epochs for the model        |
```
Comment (annie-samsung) [inline docs/application/native/guides/machine-learning/machine-learning-training.md:92]: why is the value written in brackets?
(integer) and  (file_path)

File: docs/application/native/guides/machine-learning/machine-learning-training.md [hunk 5]
```diff
@@ -74,101 +74,230 @@ ml_train_model_construct(&model);
...
+| &#xfeff;                   | save_path | (file_path)   | Model path to save parameters after a single epoch |
 
 ### Layer
 
 Layer is a component that does actual computation while managing internal trainable parameters.
-Currently, input layer and fully connected layer type are supported:
+This is how to create and add layer to the model.
```
Comment (annie-samsung) [inline docs/application/native/guides/machine-learning/machine-learning-training.md:98]:
```suggestion
The following example shows how to create and add layer to the model:
```

File: docs/application/native/guides/machine-learning/machine-learning-training.md [hunk 6]
```diff
@@ -74,101 +74,230 @@ ml_train_model_construct(&model);
...
 
 ```c
 // Create layer
 ml_train_layer_h layer;
 ml_train_layer_create(&layer, ML_TRAIN_LAYER_TYPE_FC);
 
 // configure layer
```
Comment (annie-samsung) [inline docs/application/native/guides/machine-learning/machine-learning-training.md:105]: // Configure layer

File: docs/application/native/guides/machine-learning/machine-learning-training.md [hunk 7]
```diff
@@ -74,101 +74,230 @@ ml_train_model_construct(&model);
...
-ml_train_set_property(layer, "unit=10", "activation=softmax", "bias_initializer=zeros", NULL);
+ml_train_layer_set_property(layer, "unit=10", "activation=softmax", "bias_initializer=zeros", NULL);
 
 // after adding the layer to model,
```
Comment (annie-samsung) [inline docs/application/native/guides/machine-learning/machine-learning-training.md:108]:
```suggestion
// After adding the layer to model, you do not need to destroy the layer as the ownership is transferred to the model.
```

File: docs/application/native/guides/machine-learning/machine-learning-training.md [hunk 8]
```diff
@@ -74,101 +74,230 @@ ml_train_model_construct(&model);
...
-| &#xfeff;                         |                             | global_average       | global average pooling                                                |
-| `ML_TRAIN_LAYER_TYPE_FLATTEN`    |                             |                      | Flatten layer                                                         |
-| `ML_TRAIN_LAYER_TYPE_ACTIVATION` |                             |                      | Standalone, simple activation layer                                   |
-| `ML_TRAIN_LAYER_TYPE_ADDITION`   |                             |                      | Addition layer                                                        |
-| `ML_TRAIN_LAYER_TYPE_CONCAT`     |                             |                      | Concatenation layer                                                   |
-| `ML_TRAIN_LAYER_TYPE_MULTIOUT`   |                             |                      | Multiout layer, repeats output to the number of layers                 |
-
+There are 2 types of layers.
+One type includes commonly trainable weigths and another type does not includes.
+The following are the available properties for each layer type which include commonly trainable weights.
```
Comment (annie-samsung) [inline docs/application/native/guides/machine-learning/machine-learning-training.md:115]:
```suggestion
The following are the available properties for each layer type which include commonly trainable weights;
```

File: docs/application/native/guides/machine-learning/machine-learning-training.md [hunk 9]
```diff
@@ -74,101 +74,230 @@ ml_train_model_construct(&model);
...
+
+Type | Key | Value | Default value | Description
+---------- | --- | ----- | ----------- | -----------
```
Comment (annie-samsung) [inline docs/application/native/guides/machine-learning/machine-learning-training.md:118]: Some of the values are in brackets some are not in, any specific reason?
Recommend to follow consistency.

File: docs/application/native/guides/machine-learning/machine-learning-training.md [hunk 10]
```diff
@@ -74,101 +74,230 @@ ml_train_model_construct(&model);
...
+&#xfeff;                                                     |                             | sigmoid                     |                         | Sigmoid function
+&#xfeff;                                                     |                             | relu                        |                         | Relu function
+&#xfeff;                                                     |                             | softmax                     |                         | Softmax function
+&#xfeff;                                                     | return_sequences            | (boolean)                   | false                   | Return only the last output if true, else return full output
+&#xfeff;                                                     | dropout                     | (float)                     | 0                       | Dropout rate
+
+The following are the available properties for each layer type which doesn't include (weight_initializer, bias_initializer, weight_regularizer, weight_regularizer_constant) properties.
```
Comment (annie-samsung) [inline docs/application/native/guides/machine-learning/machine-learning-training.md:203]:
```suggestion
The following are the available properties for each layer type which does not include (`weight_initializer`, `bias_initializer,` `weight_regularizer`, `weight_regularizer_constant`) properties;
```

File: docs/application/native/guides/machine-learning/machine-learning-training.md [hunk 11]
```diff
@@ -384,73 +522,113 @@ status = ml_train_model_set_optimizer(model, optimizer);
 There are two ways to create a dataset. One is from a file, and the other one is from a callback.
 In either case, you need to provide streams of tensor data and arrays of values representing the label, usually one-hot-encoded.
 
-#### Set a dataset from a file
+#### Sample, batch and epoch
+
+There are three confusing concepts need clarification: sample, batch and epoch.
```
Comment (annie-samsung) [inline docs/application/native/guides/machine-learning/machine-learning-training.md:527]:
```suggestion
This section explains the following three concepts: sample, batch and epoch.
```

File: docs/application/native/guides/machine-learning/machine-learning-training.md [hunk 12]
```diff
@@ -509,13 +684,13 @@ After training, the model must be destroyed with `ml_train_model_destroy()`.
 `ml_train_model_add_layer()`, `ml_train_set_optimizer()`, and `ml_train_set_dataset()` transfers ownership to the model.
 `layers`, `optimizers` and `dataset` that belongs to the `model` are also deleted.
 
-## Use trained model for inference
+## Use the trained model for inference (since Tizen 6.5)
```
Comment (annie-samsung) [inline docs/application/native/guides/machine-learning/machine-learning-training.md:687]: Please include since Tizen 6.5 as a note.

**Review by annie-samsung — COMMENTED** [review]
Please find my first set of comments

File: docs/application/native/guides/machine-learning/machine-learning-training.md (hunk 4 above)
Comment (zhoonit) [reply docs/application/native/guides/machine-learning/machine-learning-training.md:92]: It should be (integer) and (string) <- updated.

The value in bracket means it's a family of type not the actual value.

IMO it's (integer, boolean, string, floats) are quite common so wouldn't need additional explanation

File: docs/application/native/guides/machine-learning/machine-learning-training.md (hunk 8 above)
Comment (zhoonit) [reply docs/application/native/guides/machine-learning/machine-learning-training.md:115]: Changed to colon.

File: docs/application/native/guides/machine-learning/machine-learning-training.md (hunk 9 above)
Comment (zhoonit) [reply docs/application/native/guides/machine-learning/machine-learning-training.md:118]: discussed in https://github.com/Samsung/tizen-docs/pull/1516#discussion_r716361917

File: docs/application/native/guides/machine-learning/machine-learning-training.md (hunk 1 above)
Comment (zhoonit) [reply docs/application/native/guides/machine-learning/machine-learning-training.md]: Done
-> // Compile model - This freezes the model. Afterwards the model cannot be modified.

File: docs/application/native/guides/machine-learning/machine-learning-training.md (hunk 2 above)
Comment (zhoonit) [reply docs/application/native/guides/machine-learning/machine-learning-training.md]: Done

File: docs/application/native/guides/machine-learning/machine-learning-training.md (hunk 3 above)
Comment (zhoonit) [reply docs/application/native/guides/machine-learning/machine-learning-training.md]: Done

File: docs/application/native/guides/machine-learning/machine-learning-training.md (hunk 5 above)
Comment (zhoonit) [reply docs/application/native/guides/machine-learning/machine-learning-training.md]: Done

File: docs/application/native/guides/machine-learning/machine-learning-training.md (hunk 6 above)
Comment (zhoonit) [reply docs/application/native/guides/machine-learning/machine-learning-training.md]: Done

File: docs/application/native/guides/machine-learning/machine-learning-training.md (hunk 10 above)
Comment (zhoonit) [reply docs/application/native/guides/machine-learning/machine-learning-training.md]: Done

File: docs/application/native/guides/machine-learning/machine-learning-training.md (hunk 11 above)
Comment (zhoonit) [reply docs/application/native/guides/machine-learning/machine-learning-training.md]: Done

File: docs/application/native/guides/machine-learning/machine-learning-training.md (hunk 12 above)
Comment (zhoonit) [reply docs/application/native/guides/machine-learning/machine-learning-training.md]: Done

**Comment by zhoonit** [comment]
Please refer to https://github.com/Samsung/tizen-docs/pull/1516/commits/616f25af8b05da45d5901457c12a7358874eced6 for the changes in the first revision.

**Review by annie-samsung — COMMENTED** [review]
Please find next set of comments

File: docs/application/native/guides/machine-learning/machine-learning-training.md [hunk 13]
```diff
@@ -71,104 +71,232 @@ ml_train_model_construct(&model);
...
-| &#xfeff;                         |                             | global_max           | global max pooling                                                    |
-| &#xfeff;                         |                             | global_average       | global average pooling                                                |
-| `ML_TRAIN_LAYER_TYPE_FLATTEN`    |                             |                      | Flatten layer                                                         |
-| `ML_TRAIN_LAYER_TYPE_ACTIVATION` |                             |                      | Standalone, simple activation layer                                   |
-| `ML_TRAIN_LAYER_TYPE_ADDITION`   |                             |                      | Addition layer                                                        |
-| `ML_TRAIN_LAYER_TYPE_CONCAT`     |                             |                      | Concatenation layer                                                   |
-| `ML_TRAIN_LAYER_TYPE_MULTIOUT`   |                             |                      | Multiout layer, repeats output to the number of layers                 |
-
+There are 2 types of layers.
```
[the remark this answers is not in the document]
Comment (zhoonit) [reply docs/application/native/guides/machine-learning/machine-learning-training.md:112]: Done

File: docs/application/native/guides/machine-learning/machine-learning-training.md [hunk 14]
```diff
@@ -71,104 +71,232 @@ ml_train_model_construct(&model);
...
+One type includes commonly trainable weigths and another type does not includes.
```
[the remark this answers is not in the document]
Comment (zhoonit) [reply docs/application/native/guides/machine-learning/machine-learning-training.md:113]: Done

File: docs/application/native/guides/machine-learning/machine-learning-training.md [hunk 15]
```diff
@@ -509,13 +683,16 @@ After training, the model must be destroyed with `ml_train_model_destroy()`.
 `ml_train_model_add_layer()`, `ml_train_set_optimizer()`, and `ml_train_set_dataset()` transfers ownership to the model.
 `layers`, `optimizers` and `dataset` that belongs to the `model` are also deleted.
 
-## Use trained model for inference
+## Use the trained model for inference
+
+> [!NOTE]
+> The feature explained in this section is supported since Tizen 6.5
```
[the remark this answers is not in the document]
Comment (zhoonit) [reply docs/application/native/guides/machine-learning/machine-learning-training.md:689]: Done

File: docs/application/native/guides/machine-learning/machine-learning-training.md [hunk 16]
```diff
@@ -384,73 +521,113 @@ status = ml_train_model_set_optimizer(model, optimizer);
...
+[[label_2/1][label_2/2][label_2/3][label_2/4][label_2/5]],
+```
+
+3. Epoch
 
-To create a `dataset` from a file, a training set and a label set must be provided. A validation set and a test set can be provided if needed.
+An epoch refers to a full, exhaustive visit to a dataset.
 
-A label set is a file that has actual labels each in a line. For example:
+So, if you consider [Cifar-100](https://www.cs.toronto.edu/~kriz/cifar.html#The%20CIFAR-100%20dataset) dataset. When using 100% of training set, a sigle training epoch will be visiting 50,000 samples. If batch size is 100, it will consists of 500 batches(or iterations).
```
[the remark this answers is not in the document]
Comment (zhoonit) [reply docs/application/native/guides/machine-learning/machine-learning-training.md:554]: Could you verify below alternative?

`When using the full training set, a single training epoch will contain 50,000 samples. If batch size is 100, it will be of 500 batches(or iterations).`

I think we can't use the word `can` it's not about `it cant be`, but it is sort of the description of the fact.

File: docs/application/native/guides/machine-learning/machine-learning-training.md [hunk 17]
```diff
@@ -509,13 +683,16 @@ After training, the model must be destroyed with `ml_train_model_destroy()`.
...
-## Use trained model for inference
+## Use the trained model for inference
+
+> [!NOTE]
+> The feature explained in this section is supported since Tizen 6.5
 
 The trained model can be used for inference with [Machine Learning Inference API](machine-learning-inference.md).
-Ensure that the INI file contains the correct weight file in `save_path`.
-For example, since Tizen 6.5, you can use the trained model with single API as follows:
+Ensure that the INI file contains the correct weight file in `save_path` in `[Model]` section. The valid INI file can be made from `ml_train_model_h` with `ml_train_model_save()` if you have constructed the model with the provided api.
```
[the remark this answers is not in the document]
Comment (zhoonit) [reply docs/application/native/guides/machine-learning/machine-learning-training.md:692]: Done

File: docs/application/native/guides/machine-learning/machine-learning-training.md [hunk 18]
```diff
@@ -524,18 +701,19 @@ For example, since Tizen 6.5, you can use the trained model with single API as f
     ...
 
     ml_single_open (&single, "model_file.ini", in_info, out_info, ML_NNFW_TYPE_NNTR_INF, ML_NNFW_HW_ANY);
-    ```
+  ```
 
 You can use the trained model with with pipeline API as follows:
```
[the remark this answers is not in the document]
Comment (zhoonit) [reply docs/application/native/guides/machine-learning/machine-learning-training.md:706]: Done

File: docs/application/native/guides/machine-learning/machine-learning-training.md (hunk 15 above)
[the remark this answers is not in the document]
Comment (zhoonit) [reply docs/application/native/guides/machine-learning/machine-learning-training.md:689]: Done

File: docs/application/native/guides/machine-learning/machine-learning-training.md [hunk 19]
```diff
@@ -384,73 +521,113 @@ status = ml_train_model_set_optimizer(model, optimizer);
...
+
+A sample denotes a single pair of inputs/labels.
+
+```
+[input_1][input_2][input_3][label_1][label_2]
+```
+
+2. Batch
+
+A batch is a bundle of samples which will be fed to a single iteration. If a batch is of size 5, the batch layout, where `/*` denotes nth sample will be:
```
[the remark this answers is not in the document]
Comment (zhoonit) [reply docs/application/native/guides/machine-learning/machine-learning-training.md]: Done

**Review by annie-samsung — APPROVED** [review]
